### ISSUE TITLE
Style/ArgumentsForwarding can generate false positives

### ISSUE THREAD

**Issue by redish-developers** [rubocop/rubocop#12070]
--------

## Expected behavior

Rubocop should allow explicitly named arguments when they are used in separate invocations on the same line.

## Actual behavior

Rubocop generates a false positive.

## Steps to reproduce the problem

Example:

```ruby
def wibble(*args, &)
  wobble(*args).call(&) # <= generates Style/ArgumentsForwardingError
end
```

## RuboCop version

```
1.55.0 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [arm64-darwin22]
  - rubocop-performance 1.18.0
  - rubocop-rails 2.20.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.22.0
```

**Comment by koic** [comment]
This looks like expected behavior rather than a false positive. For example, the code below behaves as expected.

```ruby
class Baz
  def baz
    yield
  end
end

def bar(*args)
  puts args
  Baz.new
end

def foo(*, &)
  bar(*).baz(&)
end

foo('x', 'y') { puts 'z' }
```

```console
$ ruby example.rb
x
y
z
```

**Comment by amilligan** [comment]
@koic I'm confused by your comment.  The problem is not with ruby, but with rubocop.  The Style/ArgumentsForwarding cop suggests that (using your example)

```ruby
def foo(*, &)
  bar(*).baz(&)
end
```

should instead use the `(...)` argument forwarding, which would clearly not be correct.

```ruby
def foo(...)
  bar(...).baz("uh oh")
end
```

**Comment by franzliedke** [comment]
I can confirm. Here's another one:

```ruby
# Original code
def call(*args, **kwargs, &block)
  new(*args, **kwargs).call(&block)
end

# Rubocop's incorrect autocorrect
def call(...)
  new(*args, **kwargs).call(...)
end
```

That's clearly not correct.

**Comment by bbatsov** [comment]
//cc @owst

**Comment by owst** [comment]
Thanks for the ping @bbatsov, I'll try and take a look over the next couple of days, definitely something is amiss here. Apologies for the false positive - it should hopefully be easily to replicate/resolve thanks to @franzliedke and @amilligan's examples

**Comment by koic** [comment]
Ah, OK, this is a false negative because this cop's role is arguments forwarding. Somehow I missed something 😅
